Mathlib/RingTheory/Presentation.lean

```diff
@@ -229,6 +230,121 @@ def baseChange : Presentation T (T ⊗[R] S) where
   relation i := MvPolynomial.map (algebraMap R T) (P.relation i)
   span_range_relation_eq_ker := P.span_range_relation_eq_ker_baseChange
 
+end BaseChange
+
+section Composition
+
+variable {T} [CommRing T] [Algebra S T]
+variable (Q : Presentation S T) (P : Presentation R S)
+
+private noncomputable def comp_relation_aux (r : Q.rels) : MvPolynomial (Q.vars ⊕ P.vars) R :=
+  Finsupp.sum (Q.relation r)
+    (fun x j ↦ (MvPolynomial.rename Sum.inr <| P.σ j) * monomial (x.mapDomain Sum.inl) 1)
+
+private noncomputable def aux : MvPolynomial (Q.vars ⊕ P.vars) R →ₐ[R] MvPolynomial Q.vars S :=
+  aeval (Sum.elim X (MvPolynomial.C ∘ P.val))
```

Comment:
Please write a module docstring at the beginning of this section, explaining the goal and strategy. And please explain the role of `aux` (and maybe give it a docstring).

Reply:
Done, if it is not clear yet, please complain.